Cookbook/Technical-Documents/From_0%_to_100%_with_Bento.md

```diff
@@ -0,0 +1,420 @@
+                    template: appearance.tokens.icons.image(.pharmacy),
+                    backgroundColor: appearance.tokens.colors.grey50
+                )
+            )
+        )
+    }
+
+    private func renderDeliveryOption() -> Node<NodeID> {
+        return Node(
+            id: .delivery,
+            component: appearance.row3(
+                title: Localization.NHS.Prescriptions.deliveryPharmacyOptionTitle,
+                detail: Localization.NHS.Prescriptions.deliveryPharmacyOptionSubtitle,
+                image: Icon(
+                    template: appearance.tokens.icons.image(.ambulance),
+                    backgroundColor: appearance.tokens.colors.grey50
+                )
+            )
+        )
+    }
+```
+
+#### Handle tapping on a row
+Right now you should be able to see 2 rows; however, tapping on them does nothing. 
+
+A user's input (as any other event) can change ViewModel's state. As a result, a user's input also needs to go through Feedback dance in a ViewModel. To distinguish user's inputs from other events, in ViewModel there is an `enum Action`. Renderer gets a closure in the init called `observer` which takes an `Action` and passes it to a ViewModel. All you need to do is to **send** a proper `Action` into the `observer`:
+
+
+```swift
+    private func renderCollectionOption() -> Node<NodeID> {
+        return Node(
+            id: .collection,
+            component: appearance.row3(
+                title: Localization.NHS.Prescriptions.collectionPharmacyOptionTitle,
+                detail: Localization.NHS.Prescriptions.collectionPharmacyOptionSubtitle,
+                image: Icon(
+                    template: appearance.tokens.icons.image(.pharmacy),
+                    backgroundColor: appearance.tokens.colors.grey50
+                ),
+                didTap: ^Action.didTapPharmacyCollection >>> observer
+            )
+        )
+    }
+
+    private func renderDeliveryOption() -> Node<NodeID> {
+        return Node(
+            id: .delivery,
+            component: appearance.row3(
+                title: Localization.NHS.Prescriptions.deliveryPharmacyOptionTitle,
+                detail: Localization.NHS.Prescriptions.deliveryPharmacyOptionSubtitle,
+                image: Icon(
+                    template: appearance.tokens.icons.image(.ambulance),
+                    backgroundColor: appearance.tokens.colors.grey50
+                ),
+                didTap: ^Action.didTapPharmacyDelivery >>> observer
+            )
+        )
+    }
+```
+
+The `^` and `>>>` are our operators to simplify our codebase. The `^` is a lift operator which lifts single value `Action.didTapPharmacyCollection` into a function `() -> Action`. Why do we need to do so? The `>>>` composition operator needs 2 functions `(A -> B) >>> (B -> C) = (A -> C)`. In our case it's `(Void -> Action) >>> (Action -> Void) = (Void -> Void)` where `didTap` requires `Void -> Void`. Everything matches! 
```

Comment:
You could link to the [playbook page about functional operators](https://github.com/babylonhealth/ios-playbook/blob/master/Cookbook/Technical-Documents/FunctionalOperators.md) here

Reply:
will do :)